crates/simulate/src/agent/mod.rs

```diff
@@ -1,39 +1,33 @@
-use agent_channel::AgentChannel;
 use revm::primitives::AccountInfo;
 
 use crate::{
     address::Address,
-    agent::types::AgentId,
-    state::{update::UpdateData, world::World, SimState},
+    agent::agent_channel::{AgentChannelReceiver, AgentChannelSender},
+    state::{update::Update, State, World},
 };
 
 pub mod agent_channel;
-pub mod types;
 
 /// Basic traits that every `Agent` must implement in order to properly interact with an EVM and simulation.
-pub trait Agent<U: UpdateData, W: World<WorldUpdateData = U>>: Sync + Send {
+pub trait Agent<U, W>: Sync + Send
+where
+    U: Update,
+    W: World<WorldUpdate = U>,
+{
     /// Returns the address of the agent.
-    fn id(&self) -> AgentId {
-        AgentId {
-            address: Address::random(),
-            name: "Agent".into(),
-        }
-    }
+    fn address(&self) -> Address;
 
     /// Returns the account info of the agent.
     fn account_info(&self) -> AccountInfo {
         AccountInfo::default()
     }
 
     /// Executes actions against the simulation as soon as the agent is activated.
-    fn activation_step(&mut self, _state: &SimState<U, W>, _channel: AgentChannel<U>) {}
-
-    /// Executes actions against the simulation as soon as the agent is activated.
-    fn resolve_activation_step(&mut self, _state: &SimState<U, W>) {}
+    fn activation_step(&mut self, _state: &mut State<U, W>) {}
```

Comment:
Activation step now gets a mutable ref to state, which simplifies agent set-up. No need to resolve anymore since you can directly execute against the evm or world and read the outputs.